### PR TITLE
Migrates equipment slowdown to a movespeed mod

### PR DESCRIPTION
**Pull request by ninjanomnom**
More movespeed migrations, this one has some interesting side effects

:cl: ninjanomnom
fix: Held items that slow you down now affect all mobs instead of just humanoids
fix: Equipment with slowdown now affects you from any slot instead of just suits, shoes, and back slots
balance: Items that slow you down no longer only apply when there's gravity, they now apply any time you aren't floating. This means your inconvenient suit will slow you down in 0 gravity as long as you're using magboots
/:cl:

### CODE REVIEW

**Comment by cacogen** [comment]
Does the suit slowdown in zero gravity when using magboots make sense considering the magboots just stick you to the floor, they don't give your body or your clothing weight?

**Comment by Shadowflame909** [comment]
I wonder how this will effect the nukie meta of blowing up grav gen.

**Comment by carlarctg** [comment]
damn now all the hardsuits that are used by no one will now be used by actually no one due to the epic amount of slowdown they have

**Comment by ninjanomnom** [comment]
>Does that mean that taking off your hardsuit and holding it no longer works to avoid the slowdown?

No that will still work since the hands arent considered a regular equipment slot. I meant if you somehow have glasses with slowdown for example they would slow you down now with this.

>Does the suit slowdown in zero gravity when using magboots make sense considering the magboots just stick you to the floor, they don't give your body or your clothing weight?

I suppose that depends on if you consider the slowdown to be from bulkiness as weight or bulkiness as shape. Space suits *are* generally heavy but they also get in the user's way due to their design.

File: code/modules/mob/living/carbon/human/inventory.dm [hunk 1]
```diff
@@ -144,6 +144,12 @@
 
 	return not_handled //For future deeper overrides
 
+/mob/living/carbon/human/equipped_speed_mods()
+	. = ..()
+	for(var/sloties in get_all_slots())
```
Comment (actioninja) [inline code/modules/mob/living/carbon/human/inventory.dm]: hmm

File: code/modules/mob/mob.dm [hunk 2]
```diff
@@ -1229,3 +1229,17 @@
 		remove_movespeed_modifier(MOVESPEED_ID_MOB_GRAB_STATE, update=TRUE)
 	else
 		add_movespeed_modifier(MOVESPEED_ID_MOB_GRAB_STATE, update=TRUE, priority=100, override=TRUE, multiplicative_slowdown=grab_state*3, blacklisted_movetypes=FLOATING)
+
+/mob/proc/update_equipment_speed_mods()
+	var/speedies = equipped_speed_mods()
```
Comment (actioninja) [inline code/modules/mob/mob.dm]: hmmm

**Comment by actioninja** [comment]
I already know people are going to hate this but this is a worthy technical improvement and I agree with the logic of slowdown not being due to weight but due to being bulky and difficult to move quickly in.

**Pull request merged**